ci/prow-entrypoint.sh

```diff
@@ -0,0 +1,178 @@
+#!/bin/bash
+set -xeuo pipefail
+
+# Main script acting as entrypoint for all Prow jobs
+
+# Global variables
+REDIRECTOR_URL="https://rhcos-redirector.apps.art.xq1c.p1.openshiftapps.com/art/storage/releases/"
+
+# This function is used to update the /etc/passwd file within the COSA container
+# at test-time. The need for this comes from the fact that OpenShift will run a
+# container with a randomized user ID by default to enhance security. Because
+# COSA runs with an unprivileged user ("builder") instead of (container) root,
+# this presents special challenges for file and disk permissions. This particular
+# pattern was inspired by:
+# - https://cloud.redhat.com/blog/jupyter-on-openshift-part-6-running-as-an-assigned-user-id
+# - https://cloud.redhat.com/blog/a-guide-to-openshift-and-uids
+setup_user() {
+    user_id="$(id -u)"
+    group_id="$(id -g)"
+
+    cat /etc/passwd | grep -v "^builder" > /tmp/passwd
+    echo "builder:x:${user_id}:${group_id}::/home/builder:/bin/bash" >> /tmp/passwd
+    cat /tmp/passwd > /etc/passwd
+    rm /tmp/passwd
+
+    # Not strictly required, but nice for debugging.
+    id
+    whoami
+}
+
+cosa_init() {
+     # Either use the COSA_DIR prepared for us or create a temporary cosa workdir
+    cosa_dir="${COSA_DIR:-$(mktemp -d)}"
+    echo "Using $cosa_dir for build"
+    cd "$cosa_dir"
+    cosa init --transient /src
+}
+
+# Do a cosa build & cosa build-extensions only
+cosa_build() {
+    # Grab the raw value of `mutate-os-release` and use sed to convert the value
+    # to X-Y format
+    ocpver=$(rpm-ostree compose tree --print-only src/config/manifest.yaml | jq -r '.["mutate-os-release"]')
+    ocpver_mut=$(rpm-ostree compose tree --print-only src/config/manifest.yaml | jq -r '.["mutate-os-release"]' | sed 's|\.|-|')
+    prev_build_url=${REDIRECTOR_URL}/rhcos-${ocpver}/
+    # Fetch the previous build
+    cosa buildfetch --url=${prev_build_url}
+
+    # Fetch the repos corresponding to the release we are building
+    rhelver=$(rpm-ostree compose tree --print-only manifest.yaml | jq -r '.["automatic-version-prefix"]' | cut -f2 -d.)
+    curl -L "http://base-${ocpver_mut}-rhel${rhelver}.ocp.svc.cluster.local" > "src/config/ocp.repo"
+
+    cosa fetch
+    cosa build
+    cosa buildextend-extensions
```

Comment:
Maybe `cosa kola run --basic-qemu-scenarios` here?

Reply:
I ended up undoing that as this function will run in two different steps in Prow (during the build phase and during the test phase) thus we can not do kola testing here without slowing down the build/test too much.